include/magma/vk/textures/TextureManager.h

```diff
@@ -0,0 +1,39 @@
+/**
+ * @file TextureManager.h
+ * @brief Create and control textures
+ * @version 0.1
+ * @date 2021-02-12
+ */
+#pragma once
+#include <string>
+#include <unordered_map>
+
+#include "magma/vk/commandBuffer.h"
+#include "magma/vk/textures/Texture.h"
+
+class LogicalDevice;
+
+class TextureManager {
+public:
+    TextureManager(LogicalDevice &device);
+    ~TextureManager();
+
+    bool textureExists(const std::string &name) const;
+    Texture& getTexture(const std::string &name);
+
+    Texture& loadTexture(const std::string &texName, const std::string &path);
+
+    Texture& createTexture2D(const std::string &name, VkFormat format, VkExtent2D extent,
+        VkImageUsageFlags usage, VkImageAspectFlags aspectMask);
+    Texture& createTexture2D(const std::string &name, VkFormat format, uint width, uint height,
```

Comment:
there is no need in this overload.
It gives no advantages comparing to the version with VkExtent2D.
You always can pass width and height, using initializer list

Reply:
Done